lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -798,13 +798,20 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
         if (AvAttr->PlatformAgnostic == PlatformAgnosticAvailabilityKind::Unavailable) {
           // Availability for *
           if (!AvAttr->Rename.empty() && isa<ValueDecl>(D)) {
-            // NB: Don't bother getting obj-c names, we can't get one for the
             // rename
             maybePrintLeadingSpace();
             os << "SWIFT_UNAVAILABLE_MSG(\"'"
                << cast<ValueDecl>(D)->getBaseName()
                << "' has been renamed to '";
-            printEncodedString(AvAttr->Rename, false);
+            auto referencedDecl = findRenamedDecl(AvAttr, D);
+            if (referencedDecl.hasValue()) {
+              SmallString<128> scratch;
+              auto renamedObjCRuntimeName = referencedDecl.getValue()->
+              getObjCRuntimeName()->getString(scratch);
+              printEncodedString(renamedObjCRuntimeName, false);
+            } else {
+              printEncodedString(AvAttr->Rename, false);
+            }
```

Comment:
nit: This is identical to line 923-931, and almost the same as 839-847. consider refactor this into a function?

Reply:
I'm with varokas on this. I think you could push this all into `findRenamedDecl` and rename it to something like `printRenameForDecl`.